tests/test_transforms/test_embeddings/test_embedding_segment_transform.py

```diff
@@ -0,0 +1,208 @@
+    ],
+)
+@pytest.mark.smoke
+def test_backtest_full_series(example_tsds, embedding_model):
+    emb_transform = EmbeddingSegmentTransform(in_columns=["target"], embedding_model=embedding_model, out_column="emb")
+    transforms = [emb_transform]
+
+    pipeline = Pipeline(model=LinearMultiSegmentModel(), transforms=transforms, horizon=7)
+    pipeline.backtest(ts=example_tsds, metrics=[SMAPE()], n_folds=2, n_jobs=2, joblib_params=dict(backend="loky"))
+
+
+@pytest.mark.parametrize(
+    "embedding_model",
+    [
+        TS2VecEmbeddingModel(input_dims=1, n_epochs=1, output_dims=6),
+    ],
+)
+@pytest.mark.smoke
+def test_make_future(example_tsds, embedding_model):
+    emb_transform = EmbeddingSegmentTransform(in_columns=["target"], embedding_model=embedding_model, out_column="emb")
+    emb_transform.fit(example_tsds)
+
+    make_future_df = example_tsds.make_future(5, transforms=[emb_transform]).df
+    values_make_future = make_future_df.loc[:, pd.IndexSlice[:, emb_transform._get_out_columns()]].values[0]
+
+    example_tsds.transform([emb_transform])
+    ts_df = example_tsds.df
+    values_ts = ts_df.loc[:, pd.IndexSlice[:, emb_transform._get_out_columns()]].values[0]
+
+    assert np.array_equal(values_make_future, values_ts)
+
+
+@pytest.mark.parametrize(
+    "embedding_model",
+    [
+        TS2VecEmbeddingModel(input_dims=3, n_epochs=1),
+    ],
+)
+@pytest.mark.smoke
+def test_save(ts_with_exog_nan_begin, tmp_path, embedding_model):
+    transform = EmbeddingSegmentTransform(
+        in_columns=["target", "exog_1", "exog_2"], embedding_model=embedding_model, out_column="emb"
+    )
+    transform.fit(ts=ts_with_exog_nan_begin)
+
+    path = pathlib.Path(tmp_path) / "tmp.zip"
+    transform.save(path=path)
+
+
+@pytest.mark.parametrize(
+    "embedding_model",
+    [
+        TS2VecEmbeddingModel(input_dims=3, n_epochs=1),
+    ],
+)
+@pytest.mark.smoke
+def test_load(ts_with_exog_nan_begin, tmp_path, embedding_model):
+    transform = EmbeddingSegmentTransform(
+        in_columns=["target", "exog_1", "exog_2"], embedding_model=embedding_model, out_column="emb"
+    )
+    transform.fit(ts=ts_with_exog_nan_begin)
+
+    path = pathlib.Path(tmp_path) / "tmp.zip"
+    transform.save(path=path)
+    EmbeddingSegmentTransform.load(path=path)
+
+
+@pytest.mark.parametrize(
+    "output_dims, out_column, expected_out_columns",
+    [(2, "emb", ["emb_0", "emb_1"]), (3, "lag", ["lag_0", "lag_1", "lag_2"])],
+)
+def test_get_out_columns(output_dims, out_column, expected_out_columns):
+    transform = EmbeddingSegmentTransform(
+        in_columns=Mock(), embedding_model=Mock(output_dims=output_dims), out_column=out_column
+    )
+    assert sorted(expected_out_columns) == sorted(transform._get_out_columns())
+
+
+@pytest.mark.parametrize("embedding_model", [TS2VecEmbeddingModel(input_dims=3, n_epochs=1)])
+def test_second_fit_not_update_state(ts_with_exog_nan_begin, embedding_model):
```

Comment:
Can you explain what does this test? What state do you mean?

Reply:
source implementation of ts2vec do not support 2nd fit. it will not do nothing.
by the way it is not good in my opinion (but there are some hacks how to overcome this restriction)

3 ways in this situation
1. leave everything as it was 
2. remove this restriction
3. write about hacks how to overcome

Reply:
- What is a source implementation?

Reply:
in libs/ts2vec